clients/src/main/java/org/apache/kafka/clients/admin/KafkaAdminClient.java

```diff
@@ -1971,6 +2017,13 @@ public CreatePartitionsResult createPartitions(Map<String, NewPartitions> newPar
             @Override
             public void handleResponse(AbstractResponse abstractResponse) {
                 CreatePartitionsResponse response = (CreatePartitionsResponse) abstractResponse;
+                // Check for controller change
+                for (ApiError error : response.errors().values()) {
+                    if (error.error() == Errors.NOT_CONTROLLER) {
+                        runnable.shouldRequestMetadata.set(true);
```

Comment:
nit: not that big of a deal, but since we do the same thing in several places, maybe we could add a `requestMetadataUpdate` method to the runnable.